### PR TITLE
Made minor changes and bumped version to 1.3

### PR DESCRIPTION
**Pull request by cdeepakroy**
@jcfr can you please take a look?

Once we merge this, we should upload version 1.3 it to pypi for pip install

### CODE REVIEW

File: ctk_cli/module.py [hunk 1]
```diff
@@ -259,7 +258,8 @@ def isNumericVector(self):
 
     def isExternalType(self):
         """Return True iff parameter values of this type are transferred via (temporary) files"""
-        return self.typ in self.EXTERNAL_TYPES
+        return (self.typ in self.EXTERNAL_TYPES or
+                self.typ in ['file', 'directory'])
```
Comment (hmeine) [inline ctk_cli/module.py:263]: This changes the semantics of this function.  I am not saying that this is bad, but I think that at least the docstring should be adapted slightly.

**Comment by hmeine** [comment]
Except for the commented docstring, LGTM.

**Comment by cdeepakroy** [comment]
@hmeine corrected docstring. Is [this](https://github.com/commontk/ctk-cli/pull/5/files/ba2d1657d68243db56a64d57cf7e834d8736a1c5..b078ff6d13fb6a996f71e0c3968477482b3059e4#diff-c43f1fbecd4858544f6660f9afb7ff97R260) ok?

I initially thought of adding another member function to maintain backward compatibility but felt `file` and `directory` are also external types so that function should return true for them.

**Comment by hmeine** [comment]
Thanks for coming up with a simple, but apparently correct docstring.  I think it is fine.
I also checked my uses of isExternalType, and I always also handled 'file' the same.
This function always felt a little bit "fishy" to me, I think because it is not clear when one would need it.  Maybe it should also be renamed, but a clear docstring is a good step forward already, and yours might actually describe fairly well when one would need it, at last! :-)

**Comment by cdeepakroy** [comment]
One use of this function is to check whether input file or directory exists using `os.path.isfile` or `os.path.isdir` while parsing+validating command-line arguments

**Comment by cdeepakroy** [comment]
@hmeine Thanks for reviewing the code and merging :)

Could you please also upload to pypi and let me know?

**Comment by hmeine** [comment]
OK, tagged and uploaded.

I am not 100% sure that the sorting is a good idea, though.  Doesn't it throw away any semantic sorting / grouping that people may have used in their XML description?

**Comment by cdeepakroy** [comment]
Thanks for the upload to pypi!

I did the alphabetic ordering to make it easy to look up the help for a parameter when you do `--help`.

When there are a large number of parameters, i found it hard to find the parameter in the help if i preserve the same order in the xml spec.

The best thing would be to display the name of each parameter group followed by an indented list of the parameters in the group alphabetically ordered. I thought of using the [add_parameter_group](https://docs.python.org/2/library/argparse.html#argument-groups) member function of the `argparse.ArgumentParser` class to achieve this. But if the positional/indexed arguments in the xml spec are spread across groups i was not sure whether it would be nice to put them in their group or into a separate group called `positional arguments` so the order in which to provide them on command-line becomes clear to the user.

So i ended up trying to be similar to what GenerateCLP does. See a comparison of the output of `--help` of our argument_parser and GenerateCLP in this [page](https://github.com/commontk/ctk-cli/pull/4). Sorting them in alphabetical order made it easy to find a parameter in the help there.

Let me know what you think would be best. Either way it is an easy change.

**Pull request merged**